documentation/modules/upgrading/con-upgrade-custom-resources.adoc

```diff
@@ -0,0 +1,104 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrade-resources.adoc
+
+[id='con-upgrade-custom-resources-{context}']
+
+= API versioning
+
+[role="_abstract"]
+Custom resources are created as instances of APIs added by CRDs to extend Kubernetes resources.
+CRDs are installed in a Kubernetes cluster to define the schemas used to instantiate and manage the custom resources.
+
+For release {ProductVersion}, Strimzi CRDs must specify `{ApiVersion}` as the API version and `{KafkaApiVersion}` as the API path for custom resources.
```

Comment:
```suggestion
For release {ProductVersion}, Strimzi CRDs must specify `{ApiVersion}`.
```

Reply:
What does "Strimzi CRDs must specify `{ApiVersion}`" actually mean though? Is this a statement about the CRDs or the CRs? Remember that a CRD has lots of versions in it, so it's ambigous and you need to be specific.

Reply:
Honestly, when I look at it now I'm not sure what it is trying to say.

Reply:
How about if we change to this:

> For release {ProductVersion}, Strimzi custom resources must specify `{ApiVersion}` as the API version.
> Strimzi Operators access the custom resources served by the Kubernetes API server using a URL path (_API path_) which includes the API version.

Reply:
I'm still not completely sure what would be the take from this. It would make sense to say that _From release 0.23 you have to use v1beta2_. The 0.22 actually uses v1beta1 for most cases (but not all). It just adds v1beta2 to the CRDs etc.

Reply:
With the changes to the opening paragraph, maybe we can drop the first line and bring the second line up:

![image](https://user-images.githubusercontent.com/47596553/110140619-2f949880-7dcc-11eb-865e-0c04e2f7d89c.png)

Reply:
Yeah, I think this is OK for me.